packages/caravan-health/src/waste.ts

```diff
@@ -0,0 +1,183 @@
+      We can compare this ratio against the fiat charges for cross-border transactions.
+      Mastercard charges 0.6% cross-border fee for international transactions in US dollars, 
+      but if the transaction is in any other currency the fee goes up to 1%. 
+      Source : https://www.clearlypayments.com/blog/what-are-cross-border-fees-in-credit-card-payments/
+
+    Expected Range : [0, 1]
+    -> Very Poor : [1, 0.01] // More than 1% amount paid as fees. In ratio 1% is 0.01 and so on for other range
+    -> Poor : (0.01, 0.0075]
+    -> Moderate : (0.0075, 0.006]
+    -> Good : (0.006, 0.001]
+    -> Very Good : (0.001, 0)
+  */
+  feesToAmountRatio(): number {
+    let sumFeesToAmountRatio: number = 0;
+    let numberOfSendTx: number = 0;
+    const transactions = this.transactions;
+    transactions.forEach((tx: Transaction) => {
+      if (tx.isSend === true) {
+        sumFeesToAmountRatio += tx.fee / tx.amount;
+        numberOfSendTx++;
+      }
+    });
+    return sumFeesToAmountRatio / numberOfSendTx;
+  }
+
+  /*
+    Name : 
+      Spend Waste Score (S.W.S)
+
+    Definition : 
+      A score that indicates whether it is economical to spend a particular output now 
+      or wait to consolidate it later when fees could be low.
+      
+    Important Terms:
+      - Weight:
+          Transaction weight units
+      - Fee Rate:
+          The transaction's target fee rate (current fee-rate of the network)
+      - Estimated Long Term Fee Rate:
+          The long-term fee rate estimate which the wallet might need to pay 
+          to redeem remaining UTXOs.
+          Reference : https://bitcoincore.reviews/17331#l-164
+          It is the upper bound for spending the UTXO in the future.
+      - Change:
+          The cost of creating and spending a change output. It includes the fees paid 
+          on this transaction's change output plus the fees that will need to be paid 
+          to spend it later.
+      - Excess:
+          The amount by which we exceed our selection target when creating a changeless transaction, 
+          mutually exclusive with cost of change. It is extra fees paid if we don't make a change output 
+          and instead add the difference to the fees.
+      - Input Amount :
+          Sum of amount for each coin in input of the transaction
+      - Spend Amount :
+          Exact amount wanted to be spent in the transaction.
+
+    Calculation :
+      spend waste score = consolidation factor + cost of transaction
+      spend waste score = weight (fee rate - estimatedLongTermFeeRate) + change + excess
+
+    Observation :
+      Depending on the fee rate in the long term, the consolidation factor can either be positive or negative.		
+		    fee rate (current) < estimatedLongTermFeeRate (long-term fee rate)  –-> Consolidate now (-ve)
+		    fee rate (current) > estimatedLongTermFeeRate (long-term fee rate)  –-> Wait for later when fee rate go low (+ve) 
+
+  */
+  spendWasteAmount(
```

Comment:
shoudl this be called spend waste score?

Reply:
I suppose no, Since the output tells that how much amount of sats a user is saving or losing when doing this transaction now.

Let's take an example from test case,
```
      // Input UTXO Set : [0.1 BTC, 0.2 BTC, 0.3 BTC, 0.4 BTC]
      // Weight : 30 vbytes
      // Current Fee Rate : 10 sat/vbyte
      // Input Amount Sum : 10000 sats
      // Spend Amount : 8000 sats
      // Estimated Long Term Fee Rate : 15 sat/vbyte
      const weight = 30; // Estimated weight of the spending transaction
      const feeRate = 10; // Current Fee-Rate
      const inputAmountSum = 10000; // Sum of all inputs in the spending transaction
      const spendAmount = 8000; // Amount spent in the transaction
      const estimatedLongTermFeeRate = 15; // Estimated long term fee rate

      const wasteAmount = wasteMetric.spendWasteAmount(
        weight,
        feeRate,
        inputAmountSum,
        spendAmount,
        estimatedLongTermFeeRate,
      );
      expect(wasteAmount).toBe(1850);
```

Doing this transaction now, will save 1850 sats because we are estimating that in future fee-rate would be 15 instead of 10 (now). From the UI perspective, This is more appropriate score to be shown at the Preview Transactions Tab since we already have the Inputs, Expected Spend Value, Fee-Rate, Tx Weight etc.

Reply:
Ok, I think some comments and the readme might need to be updated then?

Reply:
I see! That makes sense. Thanks.